### PR TITLE
Dont catch load error when loading grpc lib in ruby gem

### PR DESCRIPTION
**Pull request by apolcyn**

### CODE REVIEW

**Comment by nicolasnoble** [comment]
LGTM. Let's see the results from distribtest.

**Comment by nicolasnoble** [comment]
Artifacts build: https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_build_artifacts/1581/

**Comment by apolcyn** [comment]
needed to fix the directory name i was checking.
retriggered a build https://grpc-testing.appspot.com/job/gRPC_build_packages/1560/

**Comment by nicolasnoble** [comment]
So it needs to be done from the top every time for it to work properly :-)

https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_build_artifacts/1583/

**Comment by nicolasnoble** [comment]
I'm not sure I'm getting the failure this time... https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_distribtest/1516/architecture=x86,language=ruby,platform=linux/console

File: src/ruby/lib/grpc/grpc.rb [hunk 1]
```diff
@@ -28,7 +28,12 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 begin
-  require_relative "#{RUBY_VERSION.sub(/\.\d$/, '')}/grpc_c"
-rescue LoadError
-  require_relative 'grpc_c'
+  ruby_version_dirname = /(\d+\.\d+)/.match(RUBY_VERSION)
```
Comment (nicolasnoble) [inline src/ruby/lib/grpc/grpc.rb]:
```
.match(RUBY_VERSION)
```

needs to be

```
.match(RUBY_VERSION).to_s()
```

**Comment by nicolasnoble** [comment]
Got it. The result of the match (MatchData) needs to be converted to a string first. Commented inline.

**Comment by apolcyn** [comment]
ah thanks :). once again https://grpc-testing.appspot.com/job/gRPC_build_artifacts/1588/console

**Comment by apolcyn** [comment]
It looks to me like all of the recent attempts failed with errors like java.IOException from hudson. 
Starting it again from the top here https://grpc-testing.appspot.com/job/gRPC_build_artifacts/1596/

**Comment by apolcyn** [comment]
It looks like distrib tests passed after getting past the infrastructure flakes. 
https://grpc-testing.appspot.com/job/gRPC_distribtest/1532/

**Comment by apolcyn** [comment]
checking the whole thing again with https://grpc-testing.appspot.com/job/gRPC_build_artifacts/1597/

**Comment by apolcyn** [comment]
https://grpc-testing.appspot.com/job/gRPC_distribtest/1533/ is passing. i think this is ready now

**Comment by nicolasnoble** [comment]
Fixes #7792, by the way.

**Pull request merged**